File: angular/src/app/service/api/model/remote-blacklist.dto.ts
export class AddNewUserToRemoteBlacklistDto {
    userId: number;
    penaltyDays: number;
    constructor(userId: number, penaltyDays: number) {
        this.userId = userId;
        this.penaltyDays = penaltyDays;
    }
}

export class GetRemoteBlacklistDto {
    id: number;
    userId: number;
    fullName: string;
    userName: string;
    penaltyDays: number;
    maxAllowedRemoteDays: number;
}

export class UpdatePenaltyDaysDto {
    id: number;
    penaltyDays: number;
    constructor(id: number, penaltyDays: number) {
        this.id = id;
        this.penaltyDays = penaltyDays;
    }
}

export class ImportRemoteBlacklistResultDto {
    successCount: number;
    failCount: number;
    failedList: string[];
}

File: aspnet-core/src/Timesheet.Core/DomainServices/Dto/RemoteBlacklistDto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Timesheet.DomainServices.Dto
{
    public class AddNewUserToRemoteBlacklistDto
    {
        public long UserId { get; set; }
        public int PenaltyDays { get; set; }
    }
    public class GetRemoteBlacklistDto
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public string FullName { get; set; }
        public string UserName { get; set; }
        public int PenaltyDays { get; set; }
    }
    public class UpdatePenaltyDaysDto
    {
        public long Id { get; set; }
        public int PenaltyDays { get; set; }
    }
    public class ImportRemoteBlacklistRowDto
    {
        public int Row { get; set; }
        public string MezonUserId { get; set; }
        public string Email { get; set; }
        public string PenaltyDaysStr { get; set; }
    }
    public class ImportRemoteBlacklistResultDto
    {
        public int SuccessCount { get; set; }
        public int FailCount { get; set; }
        public List<string> FailedList { get; set; }
    }
}
